src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosDataRetriever.java

```diff
@@ -0,0 +1,159 @@
+	// curl 'https://ncatsidg-dev.appspot.com/graphql' -H 'Accept-Encoding: gzip, deflate, br' -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Connection: keep-alive' -H 'DNT: 1' -H 'Origin: https://ncatsidg-dev.appspot.com' --data-binary '{"query":"query {  targets {    targets(top:100) {      uniprot    }  }}\n"}' --compressed
+	private static int batchSize = 1000;
+
+	@Override
+	protected void doHttpDownload(Path path, HttpClientContext context) throws Exception, HttpHostConnectException, IOException
+	{
+		int numRequests = 0;
+		boolean moreRecords = true;
+		try(FileWriter fw = new FileWriter(this.destination))
+		{
+			while (moreRecords)
+			{
+				// TODO: In the future, make FileRetriever in release-common-lib more generic so that it deals with HttpEntityEnclosingRequestBase
+				// and let subclasses optionally inject their own request object (HttpGet or HttpPost).
+				// But there just isn't time before Release 77 for that sort of elegant solution, so yes, there *is* some duplication of effort in this
+				// doHttpDownload method and the method it overrode.
+				HttpPost post = new HttpPost(PharosDataRetriever.pharosEndpoint);
+
+
+				// We will need to keep making posts until we have them all. Requests will get 100 items at a time.
+				RequestConfig config = RequestConfig.copy(RequestConfig.DEFAULT)
+											.setConnectTimeout(1000 * (int)this.timeout.getSeconds())
+											.setSocketTimeout(1000 * (int)this.timeout.getSeconds())
+											.setConnectionRequestTimeout(1000 * (int)this.timeout.getSeconds())
+											.setContentCompressionEnabled(true)
+											.build();
+				post.setConfig(config);
+				// update the query: we just increment the "skip" value each time through the loop, until we send a query that returns an empty list.
+				// We *could* use the "count" value that comes back in the first query to request one huge batch, but I think this is nicer to their server.
+				String updatedQuery = this.getQuery()
+										.replace(PharosDataRetriever.TOP_AMOUNT_TOKEN, Integer.toString(PharosDataRetriever.batchSize))
+										.replace(PharosDataRetriever.SKIP_AMOUNT_TOKEN, Integer.toString(numRequests * PharosDataRetriever.batchSize));
+				StringEntity stringEntity = new StringEntity(updatedQuery);
+				stringEntity.setContentType(PharosDataRetriever.APPLICATION_JSON);
+				post.setEntity(stringEntity);
+				post.setHeader(HttpHeaders.ACCEPT, PharosDataRetriever.APPLICATION_JSON);
+				post.setHeader(HttpHeaders.CONTENT_TYPE, PharosDataRetriever.APPLICATION_JSON);
+				int retries = this.numRetries;
+				boolean done = retries + 1 <= 0;
+				while (!done)
```

Comment:
I find the `retries + 1 <= 0` a bit confusing.  Could the condition be `retries < 0` since the condition will evaluate to the same thing?